x-pack/plugins/watcher/server/plugin.ts

```diff
@@ -45,21 +45,26 @@ export class WatcherServerPlugin implements Plugin<void, void, any, any> {
     this.log = ctx.logger.get();
   }
 
-  async setup(
-    { http, elasticsearch: elasticsearchService }: CoreSetup,
-    { licensing }: Dependencies
-  ) {
+  async setup({ http, getStartServices }: CoreSetup, { licensing }: Dependencies) {
     const router = http.createRouter();
     const routeDependencies: RouteDependencies = {
       router,
       getLicenseStatus: () => this.licenseStatus,
     };
 
-    const config = { plugins: [elasticsearchJsPlugin] };
-    const watcherESClient = elasticsearchService.createClient('watcher', config);
+    const getWatcherEsClient = async () => {
+      const coreStart = (await getStartServices())[0];
```

Comment:
What if change getStartServices function signature to `() => { core: CoreStart; plugins: PluginsStart }`?

Reply:
`const coreStart = (await getStartServices())[0];` should probably be changed to `const [coreStart] = (await getStartServices());` which is the same sugar we could use with suggested change (`const { core } = (await getStartServices());`)

Still fine with the signature change if we want to do it, but for consistency this should be done on both client and server side and therefor should be done in another PR.

Reply:
I've adopted Pierre's suggestion of `const [coreStart] = (await getStartServices());`